lib/sequent/core/event_store.rb

```diff
@@ -1,71 +1,105 @@
+require 'forwardable'
 require_relative 'event_record'
+require_relative 'helpers/functions'
 require_relative 'sequent_oj'
 
 module Sequent
   module Core
+
     class EventStore
+      include ActiveRecord::ConnectionAdapters::Quoting
+      extend Forwardable
+
       attr_accessor :configuration
+      def_delegators :@configuration, :stream_record_class, :event_record_class, :snapshot_event_class, :event_handlers
 
       def initialize(configuration = Sequent.configuration)
         self.configuration = configuration
+        @event_types = ThreadSafe::Cache.new
       end
 
       ##
       # Stores the events in the EventStore and publishes the events
       # to the registered event_handlers.
       #
-      def commit_events(command, events)
-        store_events(command, events)
-        publish_events(events, event_handlers)
+      # Events is a hash or array of pairs from `StreamRecord` to
```

Comment:
The is no parameter `events`. Also I'd don't like the fact it being a `hash` or `array`... Is that really necessary?